pkg/ingester/shipper.go

```diff
@@ -279,6 +298,45 @@ func writeShipperMetaFile(logger log.Logger, dir string, meta *shipperMeta) erro
 	enc := json.NewEncoder(f)
 	enc.SetIndent("", "\t")
 
+	if err := enc.Encode(meta); err != nil {
+		runutil.CloseWithLogOnErr(logger, f, "write meta file close")
+		return err
+	}
+	if err := f.Close(); err != nil {
+		return err
+	}
+
+	err = renameFile(logger, tmp, path)
+	if err != nil {
+		return errors.Wrap(err, "writing mimir shipped meta file")
+	}
+
+	// Keep backwards compatibility with the previous Mimir version.
+	// TODO Remove in Mimir 2.7.0
```

Comment:
Please 2.7 with 2.8 across the file. We'll announce in 2.6, so we can remove starting 2.8.